### PR TITLE
VS+Clang

### PR DESCRIPTION
**Pull request by Pentarctagon**
---

Not necessarily something I'm saying we should definitely do, but opening the PR to start a discussion about using Clang with Visual Studio on Windows rather than MSVC. The main idea being that it would mean we'd only need to deal with Clang and GCC rather than Clang, GCC, and MSVC, and that we'd be able to build with Clang on all the platforms we support.

### CODE REVIEW

**Comment by cooljeanius** [comment]
I think it makes sense to remove MSVC from the CI, since there were issues with the CI running out of resources, but I also think support for compiling with MSVC should stick around for people compiling locally just in case that's the only option they have available to them at the time

**Comment by loonycyborg** [comment]
Generally it's considered useful to have builds with as many compilers as possible because each one could have found an issue others missed. I have no love for MSVC in particular but it's still an extra compiler.

**Comment by Pentarctagon** [comment]
@cooljeanius That would be a bad idea more due to VS+MSVC being the only jobs that build on Windows, and we do need to make sure commits can build there.

@loonycyborg True, though there have also been some other large projects that switched (Chrome and Firefox being the first that come to mind). MSVC is also in a bit of a weird position in that its output is never actually distributed anywhere.  GCC and Clang are used for the packages distributed across macOS and Linux, but since the Windows packages are actually cross-compiled with GCC on Linux as well, it ends up being a bit of a strange situation where Windows devs are using a compiler (MSVC) whose output is never used by any players.

File: src/desktop/version.cpp [hunk 1]
```diff
@@ -381,7 +381,7 @@ std::string os_version()
 			}
 	}
 
-	if(v.szCSDVersion && *v.szCSDVersion) {
+	if(*v.szCSDVersion) {
```
Comment (soliton-) [inline src/desktop/version.cpp:384]: What's the reason behind this change?

Comment (Pentarctagon) [reply src/desktop/version.cpp:384]: Clang called it out as a warning, which strict mode convert to an error:
`address of array 'v.szCSDVersion' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]`

Comment (Pentarctagon) [reply src/desktop/version.cpp:384]: The other warnings beside the microsoft one that've been added to the cmake script would also need to have more targeted fixes, either by telling it to ignore the warning for specific lines (usually boost includes) or other minor code changes.

Comment (soliton-) [reply src/desktop/version.cpp:384]: Ah, it's an array. Makes sense.

Comment (CelticMinstrel) [reply src/desktop/version.cpp:384]: I see the PR was closed, but should this change be made anyway?

Comment (Pentarctagon) [reply src/desktop/version.cpp:384]: Done in af5ddbbe2ce6e42943e620250072a16d1d6202ba
